handler_test.go

```diff
@@ -240,3 +240,109 @@ func TestEchoBodyEndpoint(t *testing.T) {
 		t.Errorf("expected %s, got %s", expectedBody, actualBody)
 	}
 }
+
+func TestTextResponseEndpoint(t *testing.T) {
+	req, resp := &req, new(Response)
+
+	expectedBody := "text response"
+	req.Body = expectedBody
+
+	req.Method = "POST"
+	readTransfer(resp, *req)
+	TextResponse(resp, req)
+
+	if resp.StatusCode != StatusOK {
+		t.Errorf("expected %d, got %d", StatusOK, resp.StatusCode)
+	}
+
+	expectedStatus := StatusText(StatusOK)
+	if resp.Status != expectedStatus {
+		t.Errorf("expected %s, got %s", expectedStatus, resp.Status)
+	}
+
+	if actualBody := resp.Body; strings.Compare(actualBody, expectedBody) != 0 {
+		t.Errorf("expected %s, got %s", expectedBody, actualBody)
+	}
+}
+
+func TestHTMLResponseEndpoint(t *testing.T) {
+	req, resp := &req, new(Response)
+
+	expectedBody := "<html><body><p>HTML Response</p></body></html>"
+	req.Body = expectedBody
```

Comment:
does the request need to have the body here? or is this a test for echo requests?

Reply:
✅ I misread the specs here, request does not need body here and it's not supposed to echo a request here. Changed accordingly.